### PR TITLE
pythonPackages.transmission-rpc: 0.11 -> 3.2.2

### PR DESCRIPTION
**Pull request by eyJhb**
###### Motivation for this change
Flexget will no longer work with 0.11, and it is VERY outdated (2013 vs. 2020).
The new version is a fork of the old one, but it seems to work just fine (at least for Flexget!)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/transmission-rpc/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    yarl
+  ];
+
+  meta = with lib; {
+    description = "Python module that implements the Transmission bittorent client RPC protocol";
+    homepage = "https://pypi.python.org/pypi/transmission-rpc/";
+    license = licenses.mit;
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/transmission-rpc/default.nix:32]:
```suggestion
  };
}
```

File: pkgs/development/python-modules/transmission-rpc/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    six
+    typing-extensions
+    requests
+    yarl
+  ];
+
+  meta = with lib; {
+    description = "Python module that implements the Transmission bittorent client RPC protocol";
+    homepage = "https://pypi.python.org/pypi/transmission-rpc/";
+    license = licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/transmission-rpc/default.nix:29]: Please add yourself as a maintainer.

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -7851,7 +7851,9 @@ in {
 
   translationstring = callPackage ../development/python-modules/translationstring { };
 
-  transmissionrpc = callPackage ../development/python-modules/transmissionrpc { };
+  transmission-rpc = callPackage ../development/python-modules/transmission-rpc { };
+
+  transmissionrpc = self.transmission-rpc; # alias for compatibility 2020-02-07
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:7856]:
```suggestion
  transmission-rpc = callPackage ../development/python-modules/transmission-rpc { };
  transmissionrpc = self.transmission-rpc; # alias for compatibility 2020-02-07
```

Comment (eyJhb) [reply pkgs/top-level/python-packages.nix:7856]: This is not convention to remove the newline in between, see e.g. dnspython, etc. in the same file.

**Comment by eyJhb** [comment]
Fixed in latest commit.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112237` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>flexget</li>
<li>python37Packages.transmissionrpc (python37Packages.transmission-rpc)</li>
<li>python38Packages.transmissionrpc (python38Packages.transmission-rpc)</li>
<li>python39Packages.transmissionrpc (python39Packages.transmission-rpc)</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.transmissionrpc:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.transmissionrpc:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.transmissionrpc:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112237` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>flexget</li>
<li>python37Packages.transmissionrpc (python37Packages.transmission-rpc)</li>
<li>python38Packages.transmissionrpc (python38Packages.transmission-rpc)</li>
<li>python39Packages.transmissionrpc (python39Packages.transmission-rpc)</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.transmissionrpc:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.transmissionrpc:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.transmissionrpc:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

File: pkgs/development/python-modules/transmission-rpc/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    requests
+    yarl
+  ];
+
+  # no tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Python module that implements the Transmission bittorent client RPC protocol";
+    homepage = "https://pypi.python.org/pypi/transmission-rpc/";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/transmission-rpc/default.nix:31]:
```suggestion
    homepage = "https://pypi.org/project/transmission-rpc/";
```

File: pkgs/development/python-modules/transmission-rpc/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  propagatedBuildInputs = [
+    six
+    typing-extensions
+    requests
+    yarl
+  ];
+
+  # no tests
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/transmission-rpc/default.nix:27]:
```suggestion
  doCheck = false;
  pythonImportsCheck = [ "transmission_rpc" ];
```

**Comment by mweinelt** [comment]
Home-assistant was still using that one. And they're different packages on PyPi.

**Comment by eyJhb** [comment]
They are indeed different packages on pypi, but the difference is that the other was unmaintained. The new pypi is the maintained fork of the old (and very outdated) package. It should work (TM) just as before :)

Else home-assistent really needs to update, and it should be done upstream

**Pull request merged**